xarray/plot/utils.py

```diff
@@ -138,11 +142,14 @@ def _color_palette(cmap, n_colors):
             pal = cmap(colors_i)
         except ValueError:
             # ValueError happens when mpl doesn't like a colormap, try seaborn
-            try:
-                from seaborn import color_palette
+            if TYPE_CHECKING:
+                import seaborn as sns
+            else:
+                sns = attempt_import("seaborn")
```

Comment:
Before the ImportError was caught, maybe we should restore that functionality?

Reply:
xref to https://github.com/pydata/xarray/pull/9561/files#r1849261953

Reply:
That breaks passing a single color to a plot if seaborn is not installed. It does not get caught because we don't have a CI env that has matplotlib but not seaborn. (I first thought I was to blame (#9811) but all the better that I am not :wink:). To reproduce:

```python
import xarray as xr
air = xr.tutorial.open_dataset("air_temperature")
air.air.isel(time=0).plot.contourf(colors="k")
```

Maybe we can invert the logic? Try mpl first and if that fails try seaborn. Although that changes the logic a bit.

Reply:
@mathause you are right - I am not too proud to own up to breaking this! I am sorry about that:

[Before](https://github.com/pydata/xarray/blob/700191b9f71029073b45d44edfaa95bc092096f7/xarray/plot/utils.py#L141-L148), a missing Seaborn installation was handled gracefully:


```python

            try:
                from seaborn import color_palette


                pal = color_palette(cmap, n_colors=n_colors)
            except (ValueError, ImportError):
                # or maybe we just got a single color as a string
                cmap = ListedColormap([cmap], N=n_colors)
                pal = cmap(colors_i)
```

We could invert the logic as you suggest or just revert this to the prior behavior - try to import Seaborn and if it's not installed fall back to matplotlib. I can open a PR to get the conversation started if no one has beaten me to it!

Reply:
PR opened: #9835